kernel/src/main/java/org/kframework/unparser/ColorSetting.java

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2014-2018 K Team. All Rights Reserved.
+package org.kframework.unparser;
+
+import java.util.Map;
+
+/**
+ * @author Denis Bogdanas
+ *         Date: 10/9/13
+ */
+public enum ColorSetting {
+    OFF, ON, EXTENDED;
+
+    public ColorSetting color(boolean ttyStdout, String outputFile, Map<String, String> env) {
+        boolean colorize = outputFile == null && ttyStdout;
+        if (this == null) {
```

Comment:
This is even more confusing. This can never be null, because you can't call methods on a null pointer. It makes more sense now that you don't have that confusing instance field that is always self referential, but you can't handle the null case like this. Why was this turned into an instance method anyway?

Reply:
Hmmm, this came initially from `PrintOptions` (which used to be `PrettyPrintOptions`). I guess I'll put it back there, and it will make more sense. The idea was to set the "default setting" `ColorSetting` logic should live with the `ColorSetting` definition. But I guess this is more about defaults for the `PrintOption` than about the `ColorSetting` itself.